model/config/config.yaml

```diff
@@ -13,10 +13,50 @@ logger:
         handlers: [console]
 
 IGPreprocessor:
-    raw_data_path: ./.cache/data/instagram/ig_sample.tar.gz
-    raw_data_url: https://tweakstory.stefanmesken.info/data/ig_sample.tar.gz
-    raw_data_directory: ig_sample
+    raw_data_path: "./.cache/data/instagram/ig_sample.tar.gz"
+    raw_data_url: "https://tweakstory.stefanmesken.info/data/ig_sample.tar.gz"
+    raw_data_directory: "ig_sample"
     raw_data_group_names:
-        insta-caption-train.json: train
-        insta-caption-test1.json: eval
-        insta-caption-test2.json: test 
+        insta-caption-train.json: "train"
+        insta-caption-test1.json: "eval"
+        insta-caption-test2.json: "test"
+
+Trainer:
+    pad_token_id: 0
+    unk_token_id: 0
+    batch_size: 16
+    max_epochs: 100
+    device_name: "cpu"
+    dataloader_num_workers: 2
+
+    IGDataset:
+        cache_location: "./.cache/data/instagram/"
+        dataset_name: "ig_sample"
+        split: "train"
+        image_size: 320
+        crop_size: 320
+        image_normalization_mean:
+        - 0.485
+        - 0.456
+        - 0.406
+        image_normalization_std:
+        - 0.229
+        - 0.224
+        - 0.225
+
+    ShowAttendTell:
+        ResnetEncoder:
+            input_height: 320
+            input_width: 320
+            hidden_dim: 300
+
+        AttentionLSTMDecoder:
+            hidden_dimension: 300
+            embedding_dimension: 300
+            vocabulary_size: 100000
+            encoder_output_dimension: 2048
+            number_of_lstm_layers: 1
+            unk_token_id: 0
```

Comment:
why these token-id-related spec repeated twice?

Reply:
Because of the way I currently deal with configs. It's not ideal... and there are ways to get rid of these duplicates.

Probably the easiest way would be to split out these kinds of settings and `!include` them where needed. What do you think?

Reply:
lets leave it here if only `unk_token_id` is duplicated.